conans/util/files.py

```diff
@@ -55,6 +57,17 @@ def is_dirty(folder):
     return os.path.exists(dirty_file)
 
 
+@contextmanager
+def set_dirty_context_manager(folder, force_clean=False):
+    set_dirty(folder)
+    try:
+        yield
+        clean_dirty(folder)
+    except:
+        if force_clean:
+            clean_dirty(folder)
+
```

Comment:
Do we want to capture the exception or raise it later???

Reply:
If we are removing the `force_clean` then the try/except block is not needed, just, `yield` and `clean_dirty`, right?

Reply:
right, since an exception occurs, `clean_dirty` won't happen. I made an unit test to prove it. Of course, removing try/catch I'll need to update the test again, but the idea is the same, if something goes wrong, then keep the folder dirty.